### DISCUSSION TITLE
How to send NULL as :string in cffi

### DISCUSSION THREAD

**Discussion by not-null-nil** [cffi/cffi#316] [General]
Sorry to bother here, but I can't find helpful info.

I'm trying to use wayland-client from lisp side, this is my code:

   ```
 (define-foreign-library wayland-client
      (:unix (:or "libwayland-client.so.0.20.0" "libwayland-client.so"))
      (t (:default "libwayland-client")))
    
    
    (use-foreign-library wayland-client)
    
    (defcfun "wl_display_connect" :pointer
      (name :string))
    
    (wl-display-connect (null-pointer));;return NULL

```
Code in C, come from [here][1]
```

    #include <stdio.h>
    #include <stdlib.h>
    #include <wayland-client.h>
    
    struct wl_display *display = NULL;
    
    int main(int argc, char **argv) {
    
        display = wl_display_connect(NULL);
        if (display == NULL) {
    	fprintf(stderr, "Can't connect to display\n");
    	exit(1);
        }
        printf("connected to display\n");
    
        wl_display_disconnect(display);
        printf("disconnected from display\n");    
        exit(0);
    }
```

I have test the C one and it worked, how should the lisp one be correct?

How `wl_display_connect` been defined from [here][2]

` struct wl_display* wl_display_connect(const char *name)`

  [1]: https://jan.newmarch.name/Wayland/ProgrammingClient/
  [2]: https://wayland.freedesktop.org/docs/html/apb.html#Client-classwl__display_1a8a14a809eb2c083a806db2ee15523041

**Comment by sionescu** [comment]
Are you executing the code from within Emacs ? Maybe Emacs is using X11 and can't connect to Wayland.

**Comment by not-null-nil** [comment]
Sorry about open it as an issue first.
I have tried in terminal, and it looks work fine.
Now I will try to figure out how to run Emacs with Wayland, thanks.

**Comment by not-null-nil** [comment]
I think this can be close case I belive it is caused by lack of wayland env in emacs, and can be fixed either by manually import, open emacs in gui or others
